ocrd_anybaseocr/cli/ocrd_anybaseocr_cropping.py

```diff
@@ -44,408 +44,410 @@
-		if len(textarea) > 1:
-			textarea = self.marge_columns(textarea)
-			# print textarea
-
-		if len(textarea) > 0:
-			textarea = sorted(textarea, key=lambda x: (
-				(x[2]-x[0])*(x[3]-x[1])), reverse=True)
-			# print textarea
-			x1, y1, x2, y2 = textarea[0]
-			x1 = x1-20 if x1 > 20 else 0
-			x2 = x2+20 if x2 < width-20 else width
-			y1 = y1-40 if y1 > 40 else 0
-			y2 = y2+40 if y2 < height-40 else height
-
-			self.save_pf(base, [x1, y1, x2, y2])
-
-		return textarea
-
-	def process(self):
-		for (n, input_file) in enumerate(self.input_files):            
-			pcgts = page_from_file(self.workspace.download_file(input_file))                
-			binImg = self.workspace.resolve_image_as_pil(pcgts.get_Page().imageFilename)
-			fname = pcgts.get_Page().imageFilename
-			#fname = str(fname)
-			print("Process file: ", fname)
-			base, _ = ocrolib.allsplitext(fname)
-			binImg = ocrolib.read_image_binary(fname)
-
-			lineDetectH = []
-			lineDetectV = []
-			fpath = self.remove_rular(fname)
-			textarea, rgb, height, width = self.detect_textarea(fpath)
-			self.parameter['colSeparator'] = int(width * self.parameter['colSeparator'])
-
-			if len(textarea) > 1:
-				textarea = self.crop_area(textarea, binImg, rgb, base)
-				if len(textarea) == 0:
-					self.select_borderLine(fpath, lineDetectH, lineDetectV, base)
-			elif len(textarea) == 1 and (height*width*0.5 < (abs(textarea[0][2]-textarea[0][0]) * abs(textarea[0][3]-textarea[0][1]))):
-				x1, y1, x2, y2 = textarea[0]
-				x1 = x1-20 if x1 > 20 else 0
-				x2 = x2+20 if x2 < width-20 else width
-				y1 = y1-40 if y1 > 40 else 0
-				y2 = y2+40 if y2 < height-40 else height
-
-				self.save_pf(base, [x1, y1, x2, y2])
-			else:
-				self.select_borderLine(fpath, lineDetectH, lineDetectV, base)
-			
-			min_x ,min_y, max_x, max_y = textarea[0]
-			brd = BorderType(Coords=CoordsType("%i,%i %i,%i %i,%i %i,%i" % (min_x, min_y, max_x, min_y, max_x, max_y, min_x, max_y)))
-			pcgts.get_Page().set_Border(brd)                
-
-			ID = concat_padded(self.output_file_grp, n)
-			self.workspace.add_file(
-				ID=ID,
-				file_grp=self.output_file_grp,
-				pageId=input_file.pageId,
-				mimetype="image/png",
-				url=base + ".pf.png",
-				local_filename='%s/%s' % (self.output_file_grp, ID),
-				content=to_xml(pcgts).encode('utf-8'),
-			)                
+    def __init__(self, *args, **kwargs):
+        kwargs['ocrd_tool'] = OCRD_TOOL['tools']['ocrd-anybaseocr-crop']
+        kwargs['version'] = OCRD_TOOL['version']
+        super(OcrdAnybaseocrCropper, self).__init__(*args, **kwargs)
+
+    def write_crop_coordinate(self, base, coordinate):
+        x1, y1, x2, y2 = coordinate
+        with open(base + '-frame-pf.dat', 'w') as fp:
+            fp.write(str(x1)+"\t"+str(y1)+"\t"+str(x2-x1)+"\t"+str(y2-y1))  
+
+    def remove_rular(self, arg):
+        #base = arg.split(".")[0]
+        #img = cv2.cvtColor(arg, cv2.COLOR_RGB2BGR)
+        gray = cv2.cvtColor(arg, cv2.COLOR_RGB2GRAY)
```

Comment:
I don't think `cv2.cvtColor` is still needed here. You are already passing in an integer Numpy array via `ocrolib.pil2array`.